### PR TITLE
feat: Add configurable local address

### PR DESCRIPTION
**Pull request by jasonmead**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,21 +2,22 @@ FROM alpine:3.3
...
-    apk add ruby ruby-io-console ruby-bundler && \
-    rm -rf /var/cache/apk/* && \
-    mkdir -p /opt/fake-ec2-metadata-service
 
 WORKDIR /opt/fake-ec2-metadata-service
 COPY Gemfile Gemfile.lock ec2-metadata-service.rb /opt/fake-ec2-metadata-service/
 COPY spec /opt/fake-ec2-metadata-service/spec/
 
-RUN bundle install --deployment && \
+RUN apk update && \
```
Comment (bpholt) [inline Dockerfile]: Why did you move this `RUN` command below the `COPY` commands above?

**Comment by bpholt** [comment]
Should there be an addition to `docker-compose.yml` as well?

Also, I can't tell from the mobile web view; is this going into the `docker-for-mac` branch?

File: Dockerfile (hunk 1 above)
Comment (jasonmead) [inline Dockerfile]: So I could combine all of the run commands into one. Can't bundle install without the gemfile.

**Comment by jasonmead** [comment]
Yes, this should be targeting the docker-for-mac branch. I didn't want to the value to docker-compose because we don't know what it should be. That should be up to the user. It will fallback to the original implementation if not defined.

File: Dockerfile (hunk 1 above)
Comment (bpholt) [inline Dockerfile]: Oh, sure. I was overthinking it.

**Comment by bpholt** [comment]
The `docker-compose.yml` addition could just include the key (with no value) and it will take the value from the host environment. It seems like it is more or less required to be set to be useful.

**Comment by jasonmead** [comment]
Updated docker compose to default LOCAL_ADDR to shell LOCAL_ADDR: 113767a

File: docker-compose.yml [hunk 2]
```diff
@@ -6,3 +6,4 @@ fake-ec2-metadata-service:
     - 8169:80
   environment:
     - BIND_ADDR=0.0.0.0
+    - LOCAL_ADDR=${LOCAL_ADDR}
```
Comment (bpholt) [inline docker-compose.yml]: I think this can just be 

```
  - LOCAL_ADDR
```
